dev/src/dev/deps_graph.clj

```diff
@@ -575,3 +599,129 @@
         new-deps        (simulate-rename deps old-namespace->new-namespace)
         new-module-deps (into (sorted-set) (keys (all-module-deps-paths new-deps module)))]
     (set/difference old-module-deps new-module-deps)))
+
+(mu/defn- module->source-files :- [:set :string]
+  "Return the set of all *source* filenames (relative to the [[project-root]] directory) for a `module`."
+  [deps module]
+  (into
+   (sorted-set)
+   (comp (filter #(= (:module %) module))
+         (map :filename))
+   deps))
+
+(defn- file->namespace
+  "Infer the Clojure namespace from a core project filename.
+
+    (file->namespace \"/home/cam/metabase/enterprise/backend/src/metabase_enterprise/advanced_permissions/common.clj\")
+    ;; =>
+    metabase-enterprise.advanced-permissions.common"
+  [file]
+  (-> file
+      file->path-relative-to-project-root
+      (str/replace #"^enterprise/backend/" "")
+      (str/replace #"^(?:(?:src)|(?:test))/" "")
+      (str/replace #"\.clj[cs]?$" "")
+      (str/replace #"/" ".")
+      (str/replace #"_" "-")
+      symbol))
```

Comment:
I have something quite similar in mage. It makes me wonder about a shared mage/metabase zone. Could work well with the `#?(:bb "bb" :clj "clj")` conditional reader